src/sign-in-xplat/MainPage.xaml.cs

```diff
@@ -48,36 +53,41 @@ private async void OnSignInClicked(object sender, EventArgs e)
                     .Build();
 
                 // Sign-in user using MSAL and obtain an access token for MS Graph
-                GraphServiceClient graphClient = new GraphServiceClient(s_graphURL,
-                    new DelegateAuthenticationProvider(async (requestMessage) =>
+                async Task<AuthenticationHeaderValue> SignInUserAndGetTokenUsingMSAL()
+                {
+                    IEnumerable<IAccount> accounts = await s_publicClientApp.GetAccountsAsync().ConfigureAwait(false);
+                    IAccount firstAccount = accounts.FirstOrDefault();
+
+                    try
                     {
-                        IEnumerable<IAccount> accounts = await s_publicClientApp.GetAccountsAsync().ConfigureAwait(false);
-                        IAccount firstAccount = accounts.FirstOrDefault();
-
-                        try
-                        {
-                            // Signs in the user and obtains an Access token for MS Graph
-                            s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)
-                                                              .ExecuteAsync();
-                        }
-                        catch (MsalUiRequiredException ex)
-                        {
-                            // A MsalUiRequiredException happened on AcquireTokenSilentAsync. This indicates you need to call AcquireTokenAsync to acquire a token
-                            Debug.WriteLine($"MsalUiRequiredException: {ex.Message}");
-
-                            s_authResult = await s_publicClientApp.AcquireTokenInteractive(s_scopes)
-#if ANDROID
-                                                              .WithParentActivityOrWindow(Microsoft.Maui.Essentials.Platform.CurrentActivity)
-#endif
-                                                              .ExecuteAsync()
-                                                              .ConfigureAwait(false);
-                        }
+                        // Signs in the user and obtains an Access token for MS Graph
+                        s_authResult = await s_publicClientApp.AcquireTokenSilent(s_scopes, firstAccount)

```

Comment:
Also, been meaning to ask, can you just bring `s_scopes` local to this function now as `var graphScopes = new string[] { "user.read" };` to reduce an "out of sight" definition.

Reply:
thanks for taking the time and looking into the details @ckittel, I think that will help our users in the future.

done |  addressed from [6ae45](https://github.com/Azure-Samples/ms-identity-docs-code-dotnet/pull/17/commits/6ae45f23213e7998d086b58373a72607be1519f6)

Reply:
Closing this comment as I'll include it on the next review.